app/utils/id_generator.py

```
@@ -0,0 +1,163 @@
+        
+        Returns: # 새로운 id를 반환함
+            int: 다음 ID (1부터 시작)
+        """
+        counter_key = f"{entity_type}_id"
+        
+        # filelock을 사용한 파일 락
+        lock = FileLock(self.lock_file, timeout=10)
+        
+        with lock:
+            # 현재 카운터 읽기
+            with open(self.counters_file, 'r', encoding='utf-8') as f:
```

Ruff (0.14.11)

Unnecessary mode argument

Remove mode argument

(UP015)